### DISCUSSION TITLE
Cell Type Defining TFs

### DISCUSSION THREAD

**Discussion by Sayyam-Shah** [aertslab/scenicplus#154] [Q&A] [answered]
Hello!

Thank you for the fantastic tool.

How many I use the perturbation simulation method to identify cell type-defining transcription factors?

**Comment by SeppeDeWinter** [comment]
Hi @Sayyam-Shah 

Thank you.

I Would suggest following this tutorial: https://scenicplus.readthedocs.io/en/latest/Perturbation_simulation.html

**Reply by mason-sweat1** [reply]
Dear Seppe,

I would love to try this, but I'm hung up on one thing. The scplus_obj.uns.keys() does not contain ['lineState'], and I'm a bit confused about what this parameter would be for.

Could you perhaps tell me what an equivalent parameter would be in the scplus_obj.uns.keys() of the 10x multiome pbmc tutorial?

This would be extremely helpful, thanks!

Mason

**Reply by SeppeDeWinter** [reply, marked as answer]
Hi @mason-sweat1 

This is specific to the melanoma analysis we did in the tutorial. 
It contains genes that are differentially expressed across cell lines and melanoma states (hence the name lineState).

We just use this to limit the number of genes on which we have to model the perturbation (for the sake of running time).
You might choose to use highly variable genes, or differentially expressed genes for your system, or something else.


Ik hope this helps,

All the best,
Seppe

**Reply by mason-sweat1** [reply]
Hi Seppe,

Thank you I figured as much, but I couldn't actually find in the tutorial where lineState was generated, so I went with 'DEGs' for the RNA leiden clusters. 

I really appreciate your support with this amazing tool!

best,
Mason

**Reply by SeppeDeWinter** [reply]
Hi @mason-sweat1 

This part is indeed not present in the tutorial.
DEG's should be good to go!

Good luck with the analysis.


All the best,

Seppe
